src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -0,0 +1,197 @@
+  }
+
+  /**
+   * Gets the number of data points retrieved from tag a group.
+   *
+   * @param tagGroup The tag group to get the number of data points from
+   * @return returns the number of data points in specified group
+   */
+  public static int getNumGroupDataPoints(int tagGroup) {
+    int numDataPoints = 0;
+
+    for (int i = 0; i < ((ArrayList) tagManagers.get(tagGroup)).size(); i++) {
+      numDataPoints +=
+          ((RealTimeTagDataPointManager) ((ArrayList) tagManagers.get(tagGroup)).get(i)).getSize();
+    }
+
+    return numDataPoints;
+  }
+
+  /**
+   * Gets a new data point for each tag in group A and adds it to an internal queue in the tag's tag
+   * Manager.
+   *
+   * @param tagGroup The tag group to fetch data from.
+   */
+  public static void dataFetcher(int tagGroup) {
+    for (int i = 0; i < ((ArrayList) tagManagers.get(tagGroup)).size(); i++) {
+      // tag group and tag manager are the same size
+      TagInfo currentTag = ((TagInfo) ((ArrayList) tagGroupList.get(tagGroup)).get(i));
+      TagControl tagControl =
+          ((RealTimeTagDataPointManager) ((ArrayList) tagManagers.get(tagGroup)).get(i)).tagControl;
+      DataPoint data = getCurentTagValue(currentTag, tagControl);
+      // add new data point to the current tag manager
+      ((RealTimeTagDataPointManager) ((ArrayList) tagManagers.get(tagGroup)).get(i))
+          .addDataPoint(data);
+    }
+  }
+
+  /**
+   * Gets a tag's current value into a list of tag current values
+   *
+   * @param tag the tag to inspect
+   * @return returns a DataPoint object holding the tags current value
+   */
+  private static DataPoint getCurentTagValue(TagInfo tag, TagControl tagControl) {
+
+    String tagName = tag.getName();
+    TagType tagType = tag.getType();
+    int tagID = tag.getId();
+    String timeStampSeconds = String.valueOf(System.currentTimeMillis() / 1000);
+    DataPoint data = null;
+
+    if (tagControl != null) {
+      if (tagType == TagType.FLOAT) {
+        float val = (float) tagControl.getTagValueAsDouble();
+        data = new DataPointFloat(tagName, tagID, val, timeStampSeconds);
+      } else if (tagType == TagType.INTEGER) {
+        int val = tagControl.getTagValueAsInt();
+        data = new DataPointFloat(tagName, tagID, val, timeStampSeconds);
+      } else if (tagType == TagType.STRING) {
+        String val = tagControl.getTagValueAsString();
+        data = new DataPointString(tagName, tagID, val, timeStampSeconds);
+      } else if (tagType == TagType.BOOLEAN) {
+        boolean val = (tagControl.getTagValueAsLong() != 0);
+        data = new DataPointBoolean(tagName, tagID, val, timeStampSeconds);
+      } else if (tagType == TagType.DWORD) {
+        long val = tagControl.getTagValueAsLong();
+        data = new DataPointDword(tagName, tagID, val, timeStampSeconds);
+      }
+    } else {
+      Logger.LOG_WARN(
+          "tag control initialization failed, cannot retrieve data points from " + tag.getName());

```

Comment:
You are adding a dependency for the library for the Logger library. I think that is fine, but the repository README.md needs to be updated to reflect.

Reply:
I think it would be very confusing for this to break without any logs telling us what is going on. I added in a small readme update in a new commit